garage/tf/q_functions/discrete_mlp_q_function.py

```diff
@@ -69,5 +108,26 @@ def get_qval_sym(self, state_input, name):
         Args:
             state_input: The state input tf.Tensor to the network.
             name: Network variable scope.
+
+        Return:
+            The tf.Tensor output of Discrete MLP QFunction.
         """
-        return self.model.build(state_input, name=name)
+        with tf.variable_scope(self._variable_scope):
+            out = state_input
+            for model in self.models:
+                out = model.build(out, name=name)
+            return out
+
+    def clone(self, name):
```

Comment:
Should clone interface be in the base class?

Reply:
Yes, it will eventually be an interface in Model too.

Reply:
Please update it into the base class.